### PR TITLE
Add testing infrastructure for new codegen development.

### PR DESCRIPTION
**Pull request by vext01**
This allows us to add C tests prefixed "newcg_" which are only tested using the new codegen: i.e. when YKD_NEW_CODEGEN=1.

This allows us to have some partial testing for the new codegen without compromising testing on the old codegen.

Thoughts? comments?

### CODE REVIEW

**Comment by ltratt** [comment]
A prefix isn't great, because it means that alphabetic sorting won't put related tests next to each other. But having a `.newcg` prefix (which might be "one of many" prefixes e.g. `.O0`), for example, seems fine to me?

**Comment by vext01** [comment]
> A prefix isn't great, because it means that alphabetic sorting won't put related tests next to each other. 

When you use the new codegen you only run tests with the "newcg_" prefix.

When you use the old codegen you only run tests **without** the "newcg_" prefix.

So the sort is preserved.

**Comment by ltratt** [comment]
Oops, I meant it should be a suffix! [A test `a` now has two files: `a.c` and `newcg_a.c` so `ls` won't make clear that they're related. A suffix will solve that.]

**Comment by vext01** [comment]
Try [62b5fa2](https://github.com/ykjit/yk/pull/856/commits/62b5fa27b9a3f7d5740d2cffd70dbcc493896687)

File: tests/langtest_c.rs [hunk 1]
```diff
@@ -30,9 +30,9 @@ static NEW_CODEGEN: LazyLock<bool> = LazyLock::new(|| {
 fn correct_codegen_for_test(p: &Path) -> bool {
     let fname = p.file_name().unwrap().to_str().unwrap();
     if *NEW_CODEGEN {
-        fname.starts_with("newcg_")
+        fname.ends_with(".newcg.c")
```
[the remark this answers is not in the document]
Comment (vext01) [reply tests/langtest_c.rs:33]: Oh yeah. Urf...

How do we do this? A contains check? Too scruffy?

[the remark this answers is not in the document]
Comment (ltratt) [reply tests/langtest_c.rs:33]: I'd be fine with `.contains(".newcg.")`.

[the remark this answers is not in the document]
Comment (vext01) [reply tests/langtest_c.rs]: ffbfc994 does that.

**Comment by ltratt** [comment]
Please squash.

**Pull request merged**